src/ffi.rs

```diff
@@ -366,6 +366,37 @@ unsafe fn strlen(mut str_ptr: *const c_char) -> usize {
 }
 
 
+/// A trait for producing pointers that will always be valid in C.
+/// Rust doesn't promise what pointers does it give to ZST (https://doc.rust-lang.org/nomicon/exotic-sizes.html#zero-sized-types-zsts)
+/// This trait will give a pointer to a static variable in case the type is empty.
```

Comment:
nit: comment should be updated to mention the NULL thing (and maybe why we prefer NULL)

Reply:
ACK b7e20c5b120c42769872eb116ab4dbed2d965a2c